shuffle/cli/src/deploy.rs

```diff
@@ -53,65 +43,223 @@ fn publish_packages_as_transaction(
         .compute_topological_order()?
     {
         let module_id = module.self_id();
-        if module_id.address() == &account.address() {
-            println!("Deploying Module: {}", module_id);
-            let mut binary = vec![];
-            module.serialize(&mut binary)?;
-            let publish_txn = account.sign_with_transaction_builder(
-                factory.payload(TransactionPayload::Module(Module::new(binary))),
+        if module_id.address() != &new_account.address() {
+            println!("Skipping Module: {}", module_id);
+            continue;
+        }
+        println!("Deploying Module: {}", module_id);
+        let mut binary = vec![];
+        module.serialize(&mut binary)?;
+
+        let hash = send_module_transaction(&client, &mut new_account, binary).await?;
+
+        all_module_names.push(module_id.name().as_str().to_string());
+        all_hashes.push(hash);
+    }
+
+    confirm_txns_executed_from_hashes(&client, all_hashes).await?;
+    confirm_modules_exist(&client, all_module_names, derived_address).await
+}
+
+async fn send_module_transaction(
+    client: &DevApiClient,
+    account: &mut LocalAccount,
+    module_binary: Vec<u8>,
+) -> Result<String> {
+    let factory = TransactionFactory::new(ChainId::test());
+    let publish_txn = account.sign_with_transaction_builder(
+        factory.payload(TransactionPayload::Module(Module::new(module_binary))),
+    );
+    let bytes = bcs::to_bytes(&publish_txn)?;
+    let resp = client.post_transactions(bytes).await?;
+    let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+    let hash = get_hash_from_post_txn(json)?;
+    Ok(hash)
+}
+
+pub async fn confirm_txns_executed_from_hashes(
+    client: &DevApiClient,
+    all_hashes: Vec<String>,
+) -> Result<()> {
+    for hash in all_hashes.iter() {
+        let mut resp = client.get_transactions_by_hash(hash.as_str()).await?;
+        let mut json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+        let start = Instant::now();
+        while json["type"] == "pending_transaction" {
+            resp = client.get_transactions_by_hash(hash.as_str()).await?;
+            json = serde_json::from_str(resp.text().await?.as_str())?;
+            let duration = start.elapsed();
+            if duration > Duration::from_secs(10) {
+                break;
+            }
+        }
+        if is_execution_successful(&json)? {
+            println!("Transaction with hash {} executed successfully", hash);
+        } else {
+            println!(
+                "Transaction with hash {} didn't execute successfully:",
+                hash
             );
+            println!("{:#?}", &json);
+        }
+    }
+
+    Ok(())
+}
+
+fn is_execution_successful(json: &Value) -> Result<bool> {
+    json["success"]
+        .as_bool()
+        .ok_or_else(|| anyhow!("Unable to access success key"))
+}
+
+pub async fn confirm_modules_exist(
```

Comment:
Why do we need this at all? If `is_execution_successful` is true, we can move on. Let's ✂️ this whole method.

Reply:
Confirm module exists? I thought so too, but I got this comment from my last (outdated) replace json_rpc with DEV API PR: https://github.com/diem/diem/pull/9565/files#r738786745 . From here I worked on making another check to see if the module exists.

Reply:
Spoke to Xiao and he informed me that If txn is deploy a module and txn executed successfully, then if module does not exist, it means there is a bug. So i'll remove this extra confirm_module_exist check